packages/flutter_svg_test/test/flutter_svg_test_test.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutter_svg_test/flutter_svg_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('finds', () {
+    group('with bytesLoader', () {
+      testWidgets('asset svg', (WidgetTester widgetTester) async {
+        final SvgPicture asset = SvgPicture.asset('test/flutter_logo.svg');
+        await widgetTester.pumpWidget(
+          DefaultAssetBundle(
+            bundle: _FakeAssetBundle(),
+            child: asset,
+          ),
+        );
+
+        expect(find.svg(asset.bytesLoader), findsOneWidget);
+      });
+
+      testWidgets('network svg', (WidgetTester widgetTester) async {
+        await HttpOverrides.runZoned(() async {
+          final SvgPicture asset = SvgPicture.network('svg.dart');
+          await widgetTester.pumpWidget(asset);
+
+          expect(find.svg(asset.bytesLoader), findsOneWidget);
```

Comment:
So looking at this, it surprises me that we're not just passing in the `SvgPicture`, i.e. `find.svg(asset)`. 

Sorry I didn't see this until now. This is looking pretty close though.

Reply:
If adjust the finder accordingly:
      
```
        Finder svg(SvgPicture svg, {bool skipOffstage = true}) {
           return _SvgFinder(svg, skipOffstage: skipOffstage);
         }

      class _SvgFinder extends MatchFinder {
        _SvgFinder(this._svg, {super.skipOffstage});
      
         final SvgPicture _svg;
      
         @override
         String get description => "svg: '$_svg'";
      
         @override
         bool matches(Element candidate) {
           return candidate.widget == _svg;
         }
       }
```
For me  the following test should be valid:
```
      testWidgets('asset svg', (WidgetTester widgetTester) async {
        final SvgPicture asset = SvgPicture.asset(
          'test/flutter_logo.svg',
          width: 100,
          height: 400,
        );
        final SvgPicture asset2 = SvgPicture.asset(
          'test/flutter_logo.svg',
          width: 100,
          height: 400,
        );
        await widgetTester.pumpWidget(
          DefaultAssetBundle(
            bundle: _FakeAssetBundle(),
            child: asset,
          ),
        );
```
But unfortunately the test is invalid, because the equals method compares the object not the fields.
If I do this by hand my finder would look like this: 

```
        Finder svg(SvgPicture svg, {bool skipOffstage = true}) {
           return _SvgFinder(svg, skipOffstage: skipOffstage);
         }

      class _SvgFinder extends MatchFinder {
        _SvgFinder(this._svg, {super.skipOffstage});
      
         final SvgPicture _svg;
      
         @override
         String get description => "svg: '$_svg'";
      
         @override
         bool matches(Element candidate) {
           return  widget is SvgPicture &&
                      _svg.runtimeType == widget.runtimeType &&
                     _svg.width == widget.width &&
                     _svg.height == widget.height &&
                     _svg.fit == widget.fit &&
                     _svg.alignment == widget.alignment &&
                     _svg.bytesLoader == widget.bytesLoader &&
                     _svg.placeholderBuilder == widget.placeholderBuilder &&
                     _svg.matchTextDirection == widget.matchTextDirection &&
                     _svg.allowDrawingOutsideViewBox == widget.allowDrawingOutsideViewBox &&
                     _svg.semanticsLabel == widget.semanticsLabel &&
                     _svg.excludeFromSemantics == widget.excludeFromSemantics &&
                     _svg.clipBehavior == widget.clipBehavior &&
                     _svg.colorFilter == widget.colorFilter &&
                     _svg.theme == widget.theme;
         }
       }
```
While writing the  tests helper I orientated on the  `find.text`  and `find.image` from the flutter library.
In both they compare just with a subset of the fields. In `find.text`  with the string and  in`find.image` 
with the image provider. 
I think this was done because otherwise you would have to declare all fields in the test.
Sometimes it is just needed  to locate a widget and afterwards compare different aspects, with  `expect`.
Also sometimes it would be really hard to compare objects during an animation, where for example the double value has a huge amount of decimal place.

I would go with the current implementation.

Reply:
Ahh interesting. I would have assumed the impelementation would be something like `return widget is SvgPicture && _svg.bytesLoader = widget.bytesLoader`, but perhaps this makes more sense. If that's how this works for `Image` though I think it makes sense.